From 6fe6f64923937e725013de1a8e5f7d7b10047787 Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Fri, 17 Jan 2014 03:07:57 +0100
Subject: [PATCH 14/34] mirror: Don't close target

RH-Author: Fam Zheng <famz@redhat.com>
Message-id: <1389928083-8921-13-git-send-email-famz@redhat.com>
Patchwork-id: 56774
O-Subject: [RHEL-7 qemu-kvm PATCH 12/18] mirror: Don't close target
Bugzilla: 1041301
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

Let reference count manage target and don't call bdrv_close here.

Signed-off-by: Fam Zheng <famz@redhat.com>
Reviewed-by: Kevin Wolf <kwolf@redhat.com>
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
(cherry picked from commit f95c625ce4cb7863795fcc36502ac58a44fdb2f1)
Signed-off-by: Fam Zheng <famz@redhat.com>
---
 block/mirror.c | 1 -
 1 file changed, 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 block/mirror.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/block/mirror.c b/block/mirror.c
index 8837186..1e3337f 100644
--- a/block/mirror.c
+++ b/block/mirror.c
@@ -479,7 +479,6 @@ immediate_exit:
         }
         bdrv_swap(s->target, s->common.bs);
     }
-    bdrv_close(s->target);
     bdrv_unref(s->target);
     block_job_completed(&s->common, ret);
 }
-- 
1.7.1

